### PR TITLE
Add specific error for invalid_target error

### PR DESCRIPTION
**Pull request by itsjustriley**
### WHY are these changes introduced?

Resolves [#271](https://github.com/Shopify/develop-advanced-edits/issues/271)

When a user has an invalid store url, or the wrong permissions, when using theme dev in the cli they currently receive an ambiguous `invalid_target` error message. 
 

### WHAT is this pull request doing?

Replacing ambiguous `invalid_target` error with specific messaging. (Modified an existing error message to work in the pertinent file, added handling for 'invalid_target' specifically).

image

paired with @EvilGenius13 @AribaRajput @lukeh-shopify 

### How to test your changes?

1. Build the updated CLI `pnpm build`
2. `shopify-dev theme dev --store wrong_store`
3. Authenticate
4. Confirm updated error message is displayed in terminal

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible [documentation](https://shopify.dev) changes

### CODE REVIEW

File: packages/cli-kit/src/private/node/session/exchange.ts [hunk 1]
```diff
@@ -167,18 +167,32 @@ interface TokenRequestResult {
...
     // using a valid refresh token. When that happens, we take the user through the authentication flow.
     return new InvalidGrantError()
   }
-  if (error === 'invalid_request') {
+  if (errorMessage === 'invalid_request') {
     // There's an scenario when Identity returns "invalid_request" when exchanging an identity token.
     // This means the token is invalid. We clear the session and throw an error to let the caller know.
     return new InvalidRequestError()
   }
+  if (errorMessage === 'invalid_target') {
```
Comment (jamesmengo) [inline packages/cli-kit/src/private/node/session/exchange.ts:181]:
```ts
function tokenRequestErrorHandler(error: string) {
  if (error === 'invalid_grant') {
    // There's an scenario when Identity returns "invalid_grant" when trying to refresh the token
    // using a valid refresh token. When that happens, we take the user through the authentication flow.
    return new InvalidGrantError()
  }
  if (error === 'invalid_request') {
    // There's an scenario when Identity returns "invalid_request" when exchanging an identity token.
    // This means the token is invalid. We clear the session and throw an error to let the caller know.
    return new InvalidRequestError()
  }
  if (error === 'invalid_target') {
  // add a comment here
    return new InvalidTargetError(INVALID_TARGET_ERROR_MESSAGE)
  }
  // eslint-disable-next-line @shopify/cli/no-error-factory-functions
  return new AbortError(error)
}
```

Comment (jamesmengo) [reply packages/cli-kit/src/private/node/session/exchange.ts:181]: ^ can we try keeping these patterns as is?

Comment (jamesmengo) [reply packages/cli-kit/src/private/node/session/exchange.ts:181]: Ah ok thanks for the context! That's a great observation that I wasn't aware of :) 

As long as the message is in a constant, I'm happy :)

Comment (gonzaloriestra) [reply packages/cli-kit/src/private/node/session/exchange.ts:181]: I agree with extracting the message to a constant. And maybe there's no need to create a new error, you could just do something like `return new AbortError(INVALID_TARGET_ERROR_MESSAGE)`.

**Review by gonzaloriestra — APPROVED** [review]
LGTM, Thanks!

File: packages/cli-kit/src/private/node/session/exchange.ts [hunk 2]
```diff
@@ -179,6 +191,9 @@ function tokenRequestErrorHandler(error: string) {
     // This means the token is invalid. We clear the session and throw an error to let the caller know.
     return new InvalidRequestError()
   }
+  if (error === 'invalid_target') {
+    return new InvalidTargetError(invalidTartgetErrorMessage)
```
Comment (EvilGenius13) [inline packages/cli-kit/src/private/node/session/exchange.ts:195]:
```suggestion
    return new InvalidTargetError(invalidTargetErrorMessage)
```

**Review by EvilGenius13 — APPROVED** [review]

**Pull request merged**